autonomi/src/self_encryption/stream_encryption.rs

```diff
@@ -48,6 +48,30 @@ pub struct StreamProgressState {
     total_estimated_chunks: usize,
 }
 
+/// Creates a data iterator from a file path that reads the file in chunks
+fn create_file_data_iterator(file_path: &str) -> std::io::Result<impl Iterator<Item = Bytes>> {
```

Comment:
this sounds like a duplicated function to have.
requiring this may indicate some duplicated logic design

Reply:
Sounds very generic indeed but couldn't find one, do you have the duplicated one in mind?

Reply:
the self_encryption crate already do this job, if stay with the existing API calls